### ISSUE TITLE
[Bug]: User cannot navigate any drop down box through keyboard

### ISSUE THREAD

**Issue by pratikkarad** [storybookjs/storybook#21578]
### Describe the bug

I cannot navigate any dropdown by the keyboard.

![image](https://user-images.githubusercontent.com/32093370/224757440-0e4b1c6f-75ea-41f4-be56-deec7c961b51.png)

### To Reproduce

1. Navigate to the theme button on the toolbar by the keyboard
2. Once the button has the focus, hit Enter.
3. I cannot navigate the dropdown to change the theme

### Additional context

I observed this bug is there for all the dropdown boxes.
For example,
1. Settings button on the sidebar
![image](https://user-images.githubusercontent.com/32093370/224764248-2cceeb40-80eb-49ae-9bc7-66817893d385.png)
2. Clear background button
![image](https://user-images.githubusercontent.com/32093370/224764477-7cf5a2f2-f42b-406d-a886-b102f47f0fb7.png)
3. Size of the preview
![image](https://user-images.githubusercontent.com/32093370/224764622-7324b380-540d-4a58-8fb0-32894ffa0569.png)
4. Change the language button 
![image](https://user-images.githubusercontent.com/32093370/224764823-f9242c87-0178-413f-9c05-2a02cc735199.png)
5. Dropdown when the dock is on the right side
![image](https://user-images.githubusercontent.com/32093370/224765824-4678c40e-5a6f-454d-a07c-3e1dbac25d26.png)

**Comment by AadeshPatil** [comment]
Hey i would  like to work on this issue . can you assign me this . Thank you!

**Comment by shilman** [comment]
Thank you @AadeshPatil !!! 🙏   Please check out [how to contribute](https://storybook.js.org/docs/react/contribute/how-to-contribute) in our docs and feel free to ask questions in `#contributing` in [our Discord](https://discord.gg/storybook).

**Comment by karthiknadar1204** [comment]
@AadeshPatil ,are you still working on this issue

**Comment by SahilDahekar** [comment]
hey can i work on this issue i am a begineer ?

**Comment by kuluruvineeth** [comment]
is this issue still open? If so can you assign me this issue, thanks.

**Comment by danielmarcano** [comment]
Hi, @vanessayuenn, as commented in [this closed issue](https://github.com/storybookjs/storybook/pull/12904#issuecomment-1503070622), I would like to help with this fix. Nonetheless, as mentioned there, this will first require the creation of [a proposal](https://github.com/storybookjs/storybook/pull/12904#issuecomment-1505633904), since there's two ways to solve this:

- Through fully custom components (which will require further maintenance)
- Through a UI library that focuses on accessibility

My idea is to first create the proposal. I could do that this week, and based on that proposal's results, I could begin working on this.

In that closed issue I have mentioned that it seems like no one is currently working on this, but wanted to make sure before asking for getting this task assigned to me.

**Comment by kuluruvineeth** [comment]
> @SahilDahekar or @kuluruvineeth would either of you want to take this on? I'll assign it to one of you. Please also check out other issues tagged with [`good first issue`](https://github.com/storybookjs/storybook/labels/good%20first%20issue) or [`help wanted`](https://github.com/storybookjs/storybook/labels/help%20wanted). We will keep populating those tags. Thank you for your contribution!

@vanessayuenn please assign this to me I shall work on it, thanks.

**Comment by vanessayuenn** [comment]
hey @danielmarcano thank you for the extra context! Evidently, I had missed the conversation in the other PR. In this case, purely based on the order of events, I'll assign this issue to you. Looking forward to seeing your proposal! Please don't hesitate to reach out for help in our discord (the [`#contributing` channel](https://discord.com/channels/486522875931656193/839297503446695956) would be fitting).

@SahilDahekar @kuluruvineeth sorry for the confusion and the back-and-forth. I'd still strongly encourage y'all to check out other issues tagged with [`good first issue`](https://github.com/storybookjs/storybook/labels/good%20first%20issue) or [`help wanted`](https://github.com/storybookjs/storybook/labels/help%20wanted). There are tons of other areas within this project we could use help on. 

Thank you everyone for your interest in making Storybook better!

**Comment by danielmarcano** [comment]
Thank you for the information, and for assigning this one to me, @vanessayuenn!

I will take the #contributing channel into account.

**Comment by ssingh3856** [comment]
Is this issue still open?

**Comment by danielmarcano** [comment]
It is still open, although I am taking care of this issue.

We are discussing what the best solution / library would be in this [RFC] (https://github.com/storybookjs/storybook/discussions/22415#discussioncomment-6034710), as recommended by the Slack team, as this issue has been around for years, and we want to use a UI library focused in accessibility in order to solve it.

**Comment by daz079** [comment]
Will the Shortcuts menu also be included in this ticket? As an example, I took a quick look at StoryBook the other day, for a colleague who wants to use it for our design system, but we needed to be aware of the accessibility first:

I did some basic checks, screen reader, keyboard etc and on a [page similar to this one](https://govuk-react.github.io/govuk-react/?path=/docs/welcome--page) I noticed I could not get into the expanded list. This page is nothing to do with where I work, it's just one of the examples I looked at.

I had a look at the code and it's pretty clear what the issues are.

Just wondering if this also falls under the umbrella of this ticket?

**Comment by danielmarcano** [comment]
Yes, @daz079, this ticket would solve the accessibility issues of all drop-down / menu components.

They are currently using popper.js, which is not accessible, hence why we need to decide on a library to solve this without manually adding event listeners since that accessibility-specific code would then need to be maintained.

If you have some suggestions for the RFC it would be awesome to read them over there 😎

**Comment by Rimpyyadav** [comment]
Is this Issue resolved??

**Comment by danielmarcano** [comment]
Hello, everyone. I'm here to provide an update on this issue. I want to mention that, as discussed with @cdedreuille, the Storybook team has decided not to proceed with the creation of new components to address these accessibility concerns.

Therefore, what we discussed in this [RFC](https://github.com/storybookjs/storybook/discussions/22415) will not be realized, at least for now. Additionally, here's my closed [PR](https://github.com/storybookjs/storybook/pull/23725) where the changes related to this issue have been implemented.

With all that said, @vanessayuenn, should we close this issue?

**Comment by cdedreuille** [comment]
Hi @danielmarcano. We started working on a new branch for 8.0 projects and I believe we finally have a path to be able to improve these components. Let's discuss together as I believe there's a way to import the code you produced into 8.0. We made some experimentations with our new `Button`, `IconButton` and icon library and it seems that we finally have a good way to proceed.

For now, I would keep this issue open until we found a possible solutions for these dropdowns.

**Comment by danielmarcano** [comment]
Awesome news, @cdedreuille. Thanks for the update.

Would you like to discuss this over Discord, or should I re-open the [PR](https://github.com/storybookjs/storybook/pull/23725) so that you can revisit it, and we can get the conversation going over there?

**Comment by leagrdv** [comment]
Hey, everyone!
What's the status on this issue?

**Comment by Sidnioulz** [comment]
Hi everyone,

First of all, thanks @danielmarcano for giving this a go!

This issue should be resolved in the coming weeks through an overhaul of IconButton components that will introduce the necessary components for selection with a listbox. This is being worked on by the core team and myself as part of https://github.com/storybookjs/storybook/issues/31261. The PR can be checked [here](https://github.com/storybookjs/storybook/pull/31702).

We have the core components written and applied throughout, but have a few areas where we need to make copy and UX checks, and we still need to replace our popper tooltips with something that can close on clicks to the backdrop without closing on blur, so that it's compatible with our roving tabindex implementation for keyboard navigation.

Feel free to subscribe to the PR for updates.

**Comment by Sidnioulz** [comment]
Fixed on https://github.com/storybookjs/storybook/tree/a11y-consolidation which will be integrated into Storybook 10.
